src/geo/transform.js

```diff
@@ -1280,7 +1280,12 @@ class Transform {
         const groundAngle = Math.PI / 2 + this._pitch;
         const fovAboveCenter = this.fovAboveCenter;
 
-        const cameraToSeaLevelDistance = this._camera.position[2] * this.worldSize / Math.cos(this._pitch);
+        // Adjust distance to MSL by the minimum possible elevation visible on screen,
+        // this way the far plane is pushed further in the case of negative elevation.
+        const minElevationInPixels = this.elevation ?
+            Math.abs(this.elevation.getMinElevationBelowMSL()) * pixelsPerMeter :
+            0;
+        const cameraToSeaLevelDistance = ((this._camera.position[2] * this.worldSize) + minElevationInPixels) / Math.cos(this._pitch);
```

Comment:
Subtracting the elevation <= 0 would add to this distance, which seems to work based on my testing.

```suggestion
        const cameraToSeaLevelDistance = ((this._camera.position[2] * this.worldSize) - minElevationInPixels) / Math.cos(this._pitch);
```